sql/parser/eval.go

```diff
@@ -259,6 +263,147 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// DDate is the date Datum.
+type DDate struct {
```

Comment:
That's what I did initially and it seemed ugly. Using an embedded type is clean way to express this in Go, but I'm disappointed that I needed to create a struct in order to do that.